antarest/study/business/xpansion_management.py

```diff
@@ -0,0 +1,493 @@
+from io import BytesIO
+from typing import Optional, Literal, Union, List, cast
+
+from fastapi import HTTPException, UploadFile
+from pydantic import Field, BaseModel
+
+from antarest.core.model import JSON
+from antarest.study.model import Study
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.rawstudy.model.filesystem.folder_node import (
+    ChildNotFoundError,
+)
+from antarest.study.storage.rawstudy.model.filesystem.root.user.expansion.expansion import (
+    Expansion,
+)
+from antarest.study.storage.storage_service import StudyStorageService
+
+
+class XpansionSettingsDTO(BaseModel):
+    optimality_gap: Optional[float] = None
+    max_iteration: Optional[Union[int, Literal["inf"]]] = None
+    uc_type: Union[
+        Literal["expansion_fast"], Literal["expansion_accurate"]
+    ] = "expansion_fast"
+    master: Union[Literal["integer"], Literal["relaxed"]] = "integer"
+    yearly_weight: Optional[str] = None
+    additional_constraints: Optional[str] = Field(
+        None, alias="additional-constraints"
+    )
+    relaxed_optimality_gap: Optional[float] = Field(
+        None, alias="relaxed-optimality-gap"
+    )
+    cut_type: Optional[
+        Union[Literal["average"], Literal["yearly"], Literal["weekly"]]
+    ] = Field(None, alias="cut-type")
+    ampl_solver: Optional[str] = Field(None, alias="ampl.solver")
+    ampl_presolve: Optional[int] = Field(None, alias="ampl.presolve")
+    ampl_solve_bounds_frequency: Optional[int] = Field(
+        None, alias="ampl.solve_bounds_frequency"
+    )
+    relative_gap: Optional[float] = None
+    solver: Optional[Union[Literal["Cbc"], Literal["Coin"]]] = None
+
+
+class XpansionCandidateDTO(BaseModel):
+    # The id of the candidate is irrelevant, so it should stay hidden for the user
+    # The names should be the section titles of the file, and the id should be removed
+    name: str
+    link: str
+    annual_cost_per_mw: int = Field(alias="annual-cost-per-mw")
+    unit_size: Optional[float] = Field(None, alias="unit-size")
+    max_units: Optional[int] = Field(None, alias="max-units")
+    max_investment: Optional[float] = Field(None, alias="max-investment")
+    already_installed_capacity: Optional[int] = Field(
+        None, alias="already-installed-capacity"
+    )
+    link_profile: Optional[str] = Field(None, alias="link-profile")
+    already_installed_link_profile: Optional[str] = Field(
+        None, alias="already-installed-link-profile"
+    )
+
+
+class LinkNotFound(HTTPException):
+    def __init__(self, message: str) -> None:
+        super().__init__(HTTPStatus.NOT_FOUND, message)
+
+
+class LinkProfileNotFoundError(HTTPException):
+    def __init__(self, message: str) -> None:
+        super().__init__(HTTPStatus.NOT_FOUND, message)
+
+
+class AlreadyInstalledLinkProfileNotFoundError(Exception):
+    def __init__(self, message: str) -> None:
+        super().__init__(HTTPStatus.NOT_FOUND, message)
```

Comment:
you could have only one exception type since it indicate that a timeseries is missing in the capa dir, the message can be used to precise which one